src/main/config/config_eeprom.c

```diff
@@ -147,19 +126,18 @@ bool isEEPROMContentValid(void)
             return false;
         }
 
-        chk = updateChecksum(chk, p, record->size);
+        crc = crc16_ccitt_update(crc, p, record->size);
 
         p += record->size;
     }
 
     const configFooter_t *footer = (const configFooter_t *)p;
-    chk = updateChecksum(chk, footer, sizeof(*footer));
+    crc = crc16_ccitt_update(crc, footer, sizeof(*footer));
     p += sizeof(*footer);
-    chk = ~chk;
-    const uint8_t checkSum = *p;
+    const uint16_t checkSum = *(uint16_t *)p;
```

Comment:
I've added a 0xFFFF start value, and also changed the terminator to 0xFFFF. I haven't included the CRC itself in the CRC calculation, since this involves adding additional code to calculate the constant CRC value.